src/exam/ExamWrapper.test.jsx

```diff
@@ -114,6 +114,48 @@ describe('SequenceExamWrapper', () => {
     expect(queryByTestId('exam-api-error-component')).not.toBeInTheDocument();
   });
 
+  it('does not fetch exam data if already loaded and the sequence is not an exam', () => {
+    store.getState = () => ({
+      examState: Factory.build('examState', {
+        isLoading: false,
+      }),
+    });
+
+    render(
+      <ExamStateProvider>
+        <SequenceExamWrapper sequence={{ ...sequence, isTimeLimited: false }} courseId={courseId}>
+          <div>children</div>
+        </SequenceExamWrapper>
+      </ExamStateProvider>,
+      { store },
+    );
+    expect(getExamAttemptsData).not.toHaveBeenCalled();
+  });
+
+  it('does fetch exam data for non exam sequences if not already loaded', async () => {
+    // this would only occur if the user deeplinks directly to a non-exam sequence
+    store.getState = () => ({
+      examState: Factory.build('examState', {
+        isLoading: true,
+        getExamAttemptsData,
+      }),
+    });
+
+    render(
+      <ExamStateProvider>
+        <SequenceExamWrapper sequence={{ ...sequence, isTimeLimited: false }} courseId={courseId}>
+          <div>children</div>
+        </SequenceExamWrapper>
+      </ExamStateProvider>,
+      { store },
+    );
+
+    // can't figure out this test. For whatever reason the mock getExamAttemptsData
+    // is overriden by the actual value whenever this component is rendered.
+    // This pattern works in other tests.
+    // await waitFor(() => expect(getExamAttemptsData).toHaveBeenCalled());
```

Comment:
this one has me puzzled, I can see the mock is setup properly right before the render call and mocking other values such as isLoading or something custom works. Mocking a thunk gets rewritten somehow during the render/connect process I just can't figure out why.